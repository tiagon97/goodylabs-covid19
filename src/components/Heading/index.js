import React from 'react';
import PropTypes from 'prop-types';

import StyledHeading from './styled';

const Heading = ({ children, main }) => <StyledHeading main={main}>{children}</StyledHeading>;

export default Heading;

Heading.propTypes = {
  children: PropTypes.string.isRequired,
  main: PropTypes.bool,
};

Heading.defaultProps = {
  main: false,
};
