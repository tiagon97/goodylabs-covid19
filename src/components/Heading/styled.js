import styled from 'styled-components';

const StyledHeading = styled.h1`
  color: ${(props) => (props.main ? 'orangered' : '#808080')};
  padding: ${(props) => (props.main ? '' : '15px 0')};
  text-align: center;
`;

export default StyledHeading;
