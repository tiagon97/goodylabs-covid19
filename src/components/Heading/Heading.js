import styled from 'styled-components';

const Heading = styled.h1`
  color: ${(props) => (props.main ? 'orangered' : '#808080')};
  padding: ${(props) => (props.main ? '' : '15px 0')};
  text-align: center;
`;

export default Heading;
