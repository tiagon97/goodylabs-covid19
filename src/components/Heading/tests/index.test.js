/**
 *
 * Tests for Heading
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import Heading from '../index';

describe('<Heading />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Heading>Children</Heading>);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render not main hedaer and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Heading>Children</Heading>);
    expect(firstChild).toMatchSnapshot();
  });

  it('Should render main header and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Heading main>Children</Heading>);
    expect(firstChild).toMatchSnapshot();
  });
});
