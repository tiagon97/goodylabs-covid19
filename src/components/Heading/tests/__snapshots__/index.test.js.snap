// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> Should render main header and match the snapshot 1`] = `
<h1
  class="sc-AxjAm iyJNFV"
>
  Children
</h1>
`;

exports[`<Heading /> Should render not main hedaer and match the snapshot 1`] = `
<h1
  class="sc-AxjAm cYlDMx"
>
  Children
</h1>
`;
